tests/TestCase/ORM/Association/BelongsToManyTest.php

```diff
@@ -1223,73 +1218,77 @@ public function testSaveAssociatedEmptySetSuccess($value): void
     #[DataProvider('emptyProvider')]
     public function testSaveAssociatedEmptySetUpdateSuccess($value): void
     {
-        /** @var \Cake\ORM\Table|\PHPUnit\Framework\MockObject\MockBuilder $table */
-        $table = $this->getMockBuilder(Table::class)
-            ->getMock();
+        $table = new Table(['alias' => 'Tags']);
         $table->setSchema([]);
-        /** @var \Cake\ORM\Association\BelongsToMany|\PHPUnit\Framework\MockObject\MockObject $assoc */
-        $assoc = $this->getMockBuilder(BelongsToMany::class)
-            ->onlyMethods(['_saveTarget', 'replaceLinks'])
-            ->setConstructorArgs(['tags', ['sourceTable' => $table]])
-            ->getMock();
+        $assoc = new class ('tags', ['sourceTable' => $table]) extends BelongsToMany {
+            public bool $replaceLinksCalled = false;
+
+            public function replaceLinks(EntityInterface $entity, array $data, array $options = []): bool
+            {
+                $this->replaceLinksCalled = true;
```

Comment:
This is my crude way of making sure a method is called. Don't know if we want to keep this in generally, but at least this partially re-creates the "check" which was previously done via the `->expects()` methods from mocks.

Reply:
This is ok, but I don't think we really need to re-create this. We should just test the expected behavior of replace true and replace false.